CRM/Core/ClassLoader.php

```diff
@@ -91,6 +91,23 @@ protected function __construct() {
   }
 
   /**
+   * Requires the autoload.php generated by composer
+   *
+   * @return void
+   */
+  protected function requireComposerAutoload() {
+    $civicrm_base_path = dirname(dirname(__DIR__));
```

Comment:
That's neat.

According to http://php.net/dirname , the second argument was added in PHP 7. We'll be stuck with multiple `dirname()`s for a while.